### DISCUSSION TITLE
ComicTagger metadata

### DISCUSSION THREAD

**Discussion by mizaki** [comictagger/comictagger#595] [Ideas]
Now there is the plugin support there are two possible `comictagger_info.json` or `comictagger_cache.json` ideas:
1. Save the `genericmetadata` info.
2. Save the `cache` info.

1 seems the easiest. 
2 would require running the data through the respective talker on read to convert it to `genericmetadata` but, because the cache now holds the raw data https://github.com/comictagger/comictagger/issues/76 becomes possible.

**Comment by lordwelch** [comment]
The first option is much more difficult. That would mean creating a full comic metadata format, and people would expect comictagger to be able to read it's own format even when breaking updates happen. The cache is much simpler in most cases. The cache is only defined as a binary blob so a talker can do whatever it wants with it, and by definition it's a cache not the actual source.

**Comment by mizaki** [comment]
Thinking on it more, I still believe saving the `genericmetadata` would be easier (and maybe better). All that's needed it adding a version number to the `gmd` and then the plugin can decide how far back to be compatible with. So, if it's just a simple int; CT GMD plugin supports version 1-5, if it's higher that's a warning message etc. Down the line it may support 5-23 etc. To my mind it seem relatively easy but you have more experience so I might be missing the finer issues. (And the layout of GMD has been quite stable, a few lists to sets and the odd name change. And it doesn't seem like a viable alternative is going to be ready/released any time soon. And if it's broken by say a complete re-work, then a new plugin can be made while keeping the old one.)

For the `cache` version, I see two ways it could work:
1. The `comictagger_cache.json` (for example) if found is inserted/replace in the DB cache. This brings up a few questions: Is it expected that the user has to use a normal search so that cached data it picked up? Is the date checked against the current cache data (if it exists)? What about the 7 day stale cache check? Should this cache data have some demarcation in the DB cache? How to age this data?
2. The file cache data is processed through the correct talker and converted into `GMD`. The problem with this is there is only one active talker and so is the data ignored if that talker isn't the activate? If it's not then a new pathway to the right talker needs to be made. Even if the right active talker matches up, there still needs to be a new pathway to it for the metadata plugin. To my mind, this way seem more complex but maybe it's actually not?
